### PR TITLE
bump kubernetes-client to 6.4.0 (#500)

### LINKED ISSUE #500: migrate to kubernetes-client 6.4.0
depends on https://github.com/fabric8io/kubernetes-client/pull/4689, https://github.com/redhat-developer/intellij-common/pull/173

we're currently using fabric8 kubernetes client [5.12.2](https://github.com/fabric8io/kubernetes-client/releases/tag/v5.12.2). We should bump our client to the latest in the 6.x stream. 6.4.0 (not released yet, see https://github.com/fabric8io/kubernetes-client/releases) should include a [fix ](https://github.com/fabric8io/kubernetes-client/pull/4689) that's required for it to work properly on the IDEA platform.
Hints and instructions are documented at https://github.com/fabric8io/kubernetes-client/blob/master/doc/MIGRATION-v6.md

### PR DESCRIPTION
**Pull request by adietish**
depends on https://github.com/redhat-developer/intellij-common/pull/173
depends on https://github.com/fabric8io/kubernetes-client/pull/4689
fixes #500

### CODE REVIEW

**Comment by adietish** [comment]
For this bump to work the kubernetes-client needs to include the following PR: https://github.com/fabric8io/kubernetes-client/pull/4689

File: gradle.properties [hunk 1]
```diff
@@ -6,11 +6,11 @@ projectVersion=0.7.1-SNAPSHOT
...
 intellijPluginVersion=1.10.1
-intellijCommonVersion=1.8.0
+intellijCommonVersion=1.9.0-SNAPSHOT
 kotlin.stdlib.default.dependency = false
 kotlinVersionIdea221=1.6.21
 kotlinVersion=1.4.32
 kotlinStdlibIdea221=kotlin-stdlib
 kotlinStdlib=kotlin-stdlib
-kubernetesClientVersion=5.12.4
+kubernetesClientVersion=6.4-SNAPSHOT
```
Comment (adietish) [inline gradle.properties:15]: should be changed to 6.4.0 once it's released

**Comment by cloorc** [comment]
Will this update support `describe` command? It seems like the old `kubernetes-client` release doesn't support it yet.

**Comment by adietish** [comment]
Hi @cloorc 

thanks for chiming in, we love to get user feedback and try to match peoples need if possible.

> Will this update support `describe deployment/statefulset/...` command?
What kind of support would you like this plugin to provide? In the editor?

**Comment by adietish** [comment]
@jeffmaury, @lstocchi: please review. All other plugins depend on this to bump themselves too.

**Comment by cloorc** [comment]
> Hi @cloorc
> 
> thanks for chiming in, we love to get user feedback and try to match users needs if possible.
> 
> > Will this update support `describe deployment/statefulset/...` command?
> 
> What kind of support would you like this plugin to provide? A context menu item to the tree which opens a dedicated output like the jetbrains plugin?

Hmm, I'm developing some features like `describe deployment xxx` that have been implemented in `vscode` `kubernetes` plugin. And I ware an old `vscode` user that would use `kubernetes` plugin frequently. Now I back to `IntellijIDEA` for `java development user experience`. However, features supported by `kubernetes` are not as rich as `vscode` yet.

Just something interesting to talk about. I'm going to use native command line tools to support some kubernetes features that `kubernetes-client` hasn't provided yet.

Thanks a lot!

**Comment by adietish** [comment]
> > Hi @cloorc
> > thanks for chiming in, we love to get user feedback and try to match users needs if possible.
> > > Will this update support `describe deployment/statefulset/...` command?
> > 
> > 
> However, features supported by `kubernetes` are not as rich as `vscode` yet.

indeed, we're getting there but it takes time :)

> 
> Just something interesting to talk about. I'm going to use native command line tools to support some kubernetes features that `kubernetes-client` hasn't provided yet.

excellent idea, I also thought that this would be very handy when trying to inspect a k8s resource. 
I filed #553 and glad if you can add your thoughts in the comments. Of course any contribution is more than welcome. With or without your help, I can plan to start this in the next iteration.

> 
> Thanks a lot!

very welcome, glad if we can provide something that's useful to you and we're very thankful to get your feedback.

**Comment by adietish** [comment]
@jeffmaury, @lstocchi: could you please test this? It is required for all the other bumps in the various plugins that depend on this plugin.

**Review by lstocchi — CHANGES_REQUESTED** [review]
I see two errors if i start switching namespace but i am not sure how to replicate it as, sometimes, i have two errors, other times only one, or nothing at all...

1. having a cluster with atleast two namespaces
2. the default one does not have any pod
3. the other (`tekton-pipelines` in my case) have two pods
4. i'm in the `tekton-pipelines` -> expand minikube node -> expand pods
5. switch ns to default  
6. sometimes i face the error, if not i expand the pods node and i switch ns

In the gif below the tree broke up, in other cases it keeps working
break-k8s


The errors i get are two
the first

```
com.redhat.devtools.intellij.kubernetes.actions.UseNamespaceAction 
2023-01-23 15:19:03,640 [  58397]  ERROR - rains.concurrency.AsyncPromise - {
                elemen…ontext.name
            } must not be null 
java.lang.NullPointerException: {
                elemen…ontext.name
            } must not be null
	at com.redhat.devtools.intellij.kubernetes.tree.TreeStructure$ContextDescriptor.getLabel(TreeStructure.kt:170)
	at com.redhat.devtools.intellij.kubernetes.tree.TreeStructure$ContextDescriptor.getLabel(TreeStructure.kt:155)
	at com.redhat.devtools.intellij.kubernetes.tree.TreeStructure$Descriptor.update(TreeStructure.kt:306)
	at com.intellij.ide.util.treeView.PresentableNodeDescriptor.getUpdatedPresentation(PresentableNodeDescriptor.java:76)
	at com.intellij.ide.util.treeView.PresentableNodeDescriptor.update(PresentableNodeDescriptor.java:29)
	at com.intellij.ui.tree.StructureTreeModel$Node.update(StructureTreeModel.java:454)
	at com.intellij.ui.tree.StructureTreeModel$Node.<init>(StructureTreeModel.java:435)
	at com.intellij.ui.tree.StructureTreeModel$Node.<init>(StructureTreeModel.java:428)
	at com.intellij.ui.tree.StructureTreeModel.getValidChildren(StructureTreeModel.java:389)
	at com.intellij.ui.tree.StructureTreeModel.validateChildren(StructureTreeModel.java:298)
	at com.intellij.ui.tree.StructureTreeModel.getNode(StructureTreeModel.java:292)
	at com.intellij.ui.tree.StructureTreeModel.getChildren(StructureTreeModel.java:312)
	at com.intellij.ui.tree.AsyncTreeModel$CmdGetChildren.getNode(AsyncTreeModel.java:545)
	at com.intellij.ui.tree.AsyncTreeModel$Command.get(AsyncTreeModel.java:440)
	at com.intellij.ui.tree.AsyncTreeModel$Command.get(AsyncTreeModel.java:406)
	at com.intellij.util.concurrency.Invoker$Task.run(Invoker.java:316)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:178)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:688)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:634)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:64)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:165)
	at com.intellij.util.concurrency.Invoker.invokeSafely(Invoker.java:203)
	at com.intellij.util.concurrency.Invoker.lambda$offerSafely$0(Invoker.java:183)
	at com.intellij.util.concurrency.Invoker$Background.lambda$offer$0(Invoker.java:541)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:216)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$200(BoundedTaskExecutor.java:27)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.execute(BoundedTaskExecutor.java:195)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:213)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:184)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

the second
```
io.fabric8.kubernetes.client.KubernetesClientException: Failed to parse the kubeconfig.
	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:129)
	at io.fabric8.kubernetes.client.Config.loadFromKubeconfig(Config.java:748)
	at io.fabric8.kubernetes.client.Config.tryKubeConfig(Config.java:647)
	at io.fabric8.kubernetes.client.Config.autoConfigure(Config.java:291)
	at io.fabric8.kubernetes.client.Config.autoConfigure(Config.java:287)
	at com.redhat.devtools.intellij.kubernetes.model.client.ClientAdapter$Factory.create(ClientAdapter.kt:52)
	at com.redhat.devtools.intellij.kubernetes.model.AllContexts$2.invoke(AllContexts.kt:79)
	at com.redhat.devtools.intellij.kubernetes.model.AllContexts$2.invoke(AllContexts.kt:79)
	at com.redhat.devtools.intellij.kubernetes.model.AllContexts$client$1.invoke(AllContexts.kt:87)
	at com.redhat.devtools.intellij.kubernetes.model.AllContexts$client$1.invoke(AllContexts.kt:86)
	at com.redhat.devtools.intellij.kubernetes.model.util.ResettableLazyProperty.get(ResettableLazyProperty.kt:30)
	at com.redhat.devtools.intellij.kubernetes.model.AllContexts.getAll(AllContexts.kt:101)
	at com.redhat.devtools.intellij.kubernetes.model.AllContexts.getCurrent(AllContexts.kt:93)
	at com.redhat.devtools.intellij.kubernetes.model.ResourceModel.getCurrentContext(ResourceModel.kt:90)
	at com.redhat.devtools.intellij.kubernetes.tree.OpenShiftStructure.canContribute(OpenShiftStructure.kt:142)
	at com.redhat.devtools.intellij.kubernetes.tree.TreeStructure.getValidContributions(TreeStructure.kt:119)
	at com.redhat.devtools.intellij.kubernetes.tree.TreeStructure.createDescriptor(TreeStructure.kt:103)
	at com.intellij.ui.tree.StructureTreeModel$Node.<init>(StructureTreeModel.java:428)
	at com.intellij.ui.tree.StructureTreeModel.getValidRoot(StructureTreeModel.java:366)
	at com.intellij.ui.tree.StructureTreeModel.getRoot(StructureTreeModel.java:281)
	at com.intellij.ui.tree.StructureTreeModel.getRoot(StructureTreeModel.java:33)
	at com.intellij.ui.tree.AsyncTreeModel$CmdGetRoot.getNode(AsyncTreeModel.java:469)
	at com.intellij.ui.tree.AsyncTreeModel$Command.get(AsyncTreeModel.java:440)
	at com.intellij.ui.tree.AsyncTreeModel$Command.get(AsyncTreeModel.java:406)
	at com.intellij.util.concurrency.Invoker$Task.run(Invoker.java:316)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:178)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:688)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:634)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:64)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:165)
	at com.intellij.util.concurrency.Invoker.invokeSafely(Invoker.java:203)
	at com.intellij.util.concurrency.Invoker.promise(Invoker.java:250)
	at com.intellij.util.concurrency.Invoker.compute(Invoker.java:82)
	at com.intellij.ui.tree.AsyncTreeModel.submit(AsyncTreeModel.java:269)
	at com.intellij.ui.tree.AsyncTreeModel$1.process(AsyncTreeModel.java:48)
	at com.intellij.util.ui.tree.TreeModelAdapter.treeStructureChanged(TreeModelAdapter.java:41)
	at com.intellij.util.ui.tree.TreeModelListenerList.treeStructureChanged(TreeModelListenerList.java:61)
	at com.intellij.util.ui.tree.AbstractTreeModel.treeStructureChanged(AbstractTreeModel.java:31)
	at com.intellij.ui.tree.StructureTreeModel.invalidateInternal(StructureTreeModel.java:211)
	at com.intellij.ui.tree.StructureTreeModel.lambda$invalidate$4(StructureTreeModel.java:166)
	at com.intellij.ui.tree.StructureTreeModel.lambda$onValidThread$1(StructureTreeModel.java:116)
	at com.intellij.util.concurrency.Invoker$Wrapper.get(Invoker.java:338)
	at com.intellij.util.concurrency.Invoker$Wrapper.get(Invoker.java:329)
	at com.intellij.util.concurrency.Invoker$Task.run(Invoker.java:316)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:178)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:688)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:634)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:64)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:165)
	at com.intellij.util.concurrency.Invoker.invokeSafely(Invoker.java:203)
	at com.intellij.util.concurrency.Invoker.promise(Invoker.java:250)
	at com.intellij.util.concurrency.Invoker.compute(Invoker.java:82)
	at com.intellij.util.concurrency.Invoker.invoke(Invoker.java:120)
	at com.intellij.ui.tree.StructureTreeModel.onValidThread(StructureTreeModel.java:114)
	at com.intellij.ui.tree.StructureTreeModel.invalidate(StructureTreeModel.java:166)
	at com.redhat.devtools.intellij.kubernetes.tree.TreeUpdater.invalidateRoot(TreeUpdater.kt:132)
	at com.redhat.devtools.intellij.kubernetes.tree.TreeUpdater.invalidatePath(TreeUpdater.kt:120)
	at com.redhat.devtools.intellij.kubernetes.tree.TreeUpdater.invalidatePaths(TreeUpdater.kt:114)
	at com.redhat.devtools.intellij.kubernetes.tree.TreeUpdater.modified$lambda$8(TreeUpdater.kt:99)
	at com.intellij.util.concurrency.Invoker$Wrapper.get(Invoker.java:338)
	at com.intellij.util.concurrency.Invoker$Wrapper.get(Invoker.java:329)
	at com.intellij.util.concurrency.Invoker$Task.run(Invoker.java:316)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:178)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:688)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:634)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:64)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:165)
	at com.intellij.util.concurrency.Invoker.invokeSafely(Invoker.java:203)
	at com.intellij.util.concurrency.Invoker.lambda$offerSafely$0(Invoker.java:183)
	at com.intellij.util.concurrency.Invoker$Background.lambda$offer$0(Invoker.java:541)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:216)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$200(BoundedTaskExecutor.java:27)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.execute(BoundedTaskExecutor.java:195)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:213)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:184)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: com.fasterxml.jackson.databind.JsonMappingException: while scanning a simple key
 in 'reader', line 564, column 5:
        namespa
        ^
could not find expected ':'
 in 'reader', line 564, column 12:
        namespa
               ^

 at [Source: (StringReader); line: 563, column: 76] (through reference chain: io.fabric8.kubernetes.api.model.Config["contexts"]->java.util.ArrayList[58]->io.fabric8.kubernetes.api.model.NamedContext["context"])
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:402)
	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:361)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.wrapAndThrow(BeanDeserializerBase.java:1826)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:316)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:177)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer._deserializeFromArray(CollectionDeserializer.java:359)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:244)
	at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:28)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:129)
	at io.fabric8.kubernetes.model.jackson.SettableBeanPropertyDelegate.deserializeAndSet(SettableBeanPropertyDelegate.java:134)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:314)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:177)
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:323)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4730)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3677)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3645)
	at io.fabric8.kubernetes.client.internal.KubeConfigUtils.parseConfigFromString(KubeConfigUtils.java:48)
	at io.fabric8.kubernetes.client.Config.loadFromKubeconfig(Config.java:683)
	... 80 more
Caused by: com.fasterxml.jackson.dataformat.yaml.snakeyaml.error.MarkedYAMLException: while scanning a simple key
 in 'reader', line 564, column 5:
        namespa
        ^
could not find expected ':'
 in 'reader', line 564, column 12:
        namespa
               ^

 at [Source: (StringReader); line: 563, column: 76]
	at com.fasterxml.jackson.dataformat.yaml.snakeyaml.error.MarkedYAMLException.from(MarkedYAMLException.java:28)
	at com.fasterxml.jackson.dataformat.yaml.YAMLParser.nextToken(YAMLParser.java:423)
	at com.fasterxml.jackson.core.JsonParser.nextFieldName(JsonParser.java:1065)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:321)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:177)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:129)
	at io.fabric8.kubernetes.model.jackson.SettableBeanPropertyDelegate.deserializeAndSet(SettableBeanPropertyDelegate.java:134)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:314)
	... 94 more
Caused by: while scanning a simple key
 in 'reader', line 564, column 5:
        namespa
        ^
could not find expected ':'
 in 'reader', line 564, column 12:
        namespa
               ^

	at org.yaml.snakeyaml.scanner.ScannerImpl.removePossibleSimpleKey(ScannerImpl.java:575)
	at org.yaml.snakeyaml.scanner.ScannerImpl.fetchStreamEnd(ScannerImpl.java:646)
	at org.yaml.snakeyaml.scanner.ScannerImpl.fetchMoreTokens(ScannerImpl.java:357)
	at org.yaml.snakeyaml.scanner.ScannerImpl.checkToken(ScannerImpl.java:263)
	at org.yaml.snakeyaml.parser.ParserImpl$ParseBlockMappingKey.produce(ParserImpl.java:662)
	at org.yaml.snakeyaml.parser.ParserImpl.peekEvent(ParserImpl.java:185)
	at org.yaml.snakeyaml.parser.ParserImpl.getEvent(ParserImpl.java:195)
	at com.fasterxml.jackson.dataformat.yaml.YAMLParser.nextToken(YAMLParser.java:419)
	... 100 more
```

i want to clarify that the kubeconfig is always the same and i don't change anything manually. I can also see that the kubeconfig is not actually broken because if i restart the plugin without changing anything, it start re-working

**Review by lstocchi — APPROVED** [review]
Tried again with an ocp cluster. Everything works fine, probably my env is broken

**Comment by adietish** [comment]
@lstocchi thanks!
asked @olkornii to test ITs against this since this requires publishing to marketplace (so that all our plugins that depend on this plugin can reference it)

**Comment by adietish** [comment]
@olkornii ran IT tests and found a few minor bugs in the editor (ex. when editing a cluster resource you get a pull notification right when the editor is opened). We'll not tackle those since the editor is overhauled for the support of multiple resources.

**Pull request merged**